packages/typescript/src/index.ts

```diff
@@ -34,14 +34,17 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
     buildStart() {
       emitParsedOptionsErrors(ts, this, parsedOptions);
 
-      program = createWatchProgram(ts, this, {
-        formatHost,
-        resolveModule,
-        parsedOptions,
-        writeFile(fileName, data) {
-          emittedFiles.set(fileName, data);
-        }
-      });
+      // Fixes a memory leak https://github.com/rollup/plugins/issues/322
+      if (!program) {
+        program = createWatchProgram(ts, this, {
```

Comment:
@NotWoods how can we best add a test for this to protect against regressions?

Reply:
Unfortunately I'm not sure how to test for memory leaks short of re-running the test many times (which would have a large impact on our CI times). I think we should merge this in for now without an added test.

Reply:
right on.